resources/zh/cemu.po

```diff
@@ -452,76 +555,36 @@ msgstr "设置CPU模拟模式"
 
 #: GameProfileWindow.cpp
 msgid "Thread quantum"
-msgstr "Thread quantum"
+msgstr "线程量子长度为"
 
 #: GameProfileWindow.cpp
 msgid ""
 "EXPERT OPTION\n"
 "Set the maximum thread slice runtime (in virtual cycles)"
 msgstr ""
-"专家选项\n"
-"Set the maximum thread slice runtime (in virtual cycles)"
+"专业选项 ⚠\n"
+"线程量子是线程在中断并切换运行的程序前，线程留给程序使用的最大时钟滴答数（以"
+"虚拟的时钟滴答为单位，一个滴答包括若干个时钟周期）"
```

Comment:
This translation still does not convince me fully.
I wasn't aware of the difference between "clock cycle" and "clock tick", so I would stick with the first one since the word "cycles" is used in the original string (and remove the explanation of the difference between tick and cycle).
I would not use the term "program" for the explanation since Cemu runs only one game at a time (and therefore does not have a concept of "program"). I would rephrase it as "the maximum amount of time in virtual clock cycles for which a thread is allowed to run before it gets paused to let another thread do its work".

Reply:
Yes. Because in Cemu, the original text here is "Thread quantum 20000 cycles", in order to make it more fluent in Chinese, I added is to make it conform to Chinese reading habits. If you think this is inappropriate, it can be corrected, but will not read fluent.

Reply:
In the explanation you sent me earlier, you said "'cycles' here refers to the'clock ticks' mentioned in the explanation above." Regarding this, the result I found is that they are indeed different, the search results found as follows:

> clock ticks refer to the main system clock. It is the smallest unit of time recognized by the device. clock cycle is the time taken for a full processor pulse to complete. this u can recognize by your cpu cpeed given in Hz. a 2GHz processor performs 2,000,000,000 clock cycles per second.

This is why I translate it as clock ticks.
For the latter, my previous understanding was that the thread needs to suspend Cemu's work to complete other Windows programs, and leave Cemu a certain number of clock ticks, so I used the term program.
Is there a problem with my understanding? I will change it later.

Reply:
What do you think of this translation?
线程时间片长度为 xxxx 个周期
专业选项 ⚠
设置线程（以时钟滴答为单位）运行时间片的最大值
I asked some friends, Wikipedia mentioned:

The period of time for which a process is allowed to run in a preemptive multitasking system is generally called the time slice or quantum. The scheduler is run once every time slice to choose the next process to run. The length of each time slice can be critical to balancing system performance vs process responsiveness-if the time slice is too short then the scheduler will consume too much processing time, but if the time slice is too long, processes will take longer to respond to input.

An interrupt is scheduled to allow the operating system kernel to switch between processes when their time slices expire, effectively allowing the processor's time to be shared between a number of tasks, giving the illusion that it is dealing with these tasks in parallel (simultaneously). The operating system which controls such a design is called a multi-tasking system.

So translate quantum into time slice, the more commonly used term.

Reply:
> In the explanation you sent me earlier, you said "'cycles' here refers to the 'clock ticks' mentioned in the explanation above." Regarding this, the result I found is that they are indeed different

Yes, in fact initially I wasn't aware of this difference. But the key thing to understand is that Cemu says "**virtual** clock cycles", meaning the cycles of the emulated CPU. Therefore it's not related to OS ticks or other programs in the system.

> For the latter, my previous understanding was that the thread needs to suspend Cemu's work to complete other Windows programs, and leave Cemu a certain number of clock ticks, so I used the term program.

When I say thread, I refer to a piece of code running in parallel. Cemu needs to emulate the behavior of the Wii U scheduler, which needs to execute several threads concurrently (these threads are spawned by the running game) but only has 3 cores (which are emulated by Cemu) to do it.

In your Wikipedia quote you have this sentence:
> An interrupt is scheduled to allow the **operating system kernel** to switch between **processes** when their time slices expire

In Cemu context, that would be:

> An interrupt is scheduled to allow the **Cemu scheduler** to switch between **threads of the running game** when their time slices expire

I hope this is a bit more clear now.

Reply:
> 线程时间片长度为 xxxx 个周期
专业选项 ⚠
设置线程（以时钟滴答为单位）运行时间片的最大值

I would still use "quantum" in the first line. Also, you must specify that cycles are **virtual** (if you want, you can rephrase it as "cycles of the emulated CPU").
Use always "cycles" instead of "ticks" because that's the term chosen by Cemu devs (and they know better than us how Cemu works internally).

Reply:
Okay, I think I probably understand. I have uploaded a new translation, which also fixes a few minor issues that were not discovered before. If you have time, please take a look.